grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -615,18 +615,17 @@ class OrderService {
                         if (productSource) {
                             orderItem.productSupplier = productSource
                         }
-                    } else if (supplierCode && manufacturer && manufacturerCode) {
-                        if (Organization.get(manufacturer)) {
-                            Organization supplier = Organization.get(supplierId)
-                            def supplierParams = [manufacturer: manufacturer,
-                                                  product: product,
-                                                  supplierCode: supplierCode,
-                                                  manufacturerCode: manufacturerCode,
-                                                  supplier: supplier,
-                                                  sourceName: sourceName]
-                            ProductSupplier productSupplier = productSupplierDataService.getOrCreateNew(supplierParams)
-                            orderItem.productSupplier = productSupplier
-                        }
+                    } else {
+                        Organization supplier = Organization.get(supplierId)
+                        Organization manufacturer = Organization.findByName(manufacturerName)
```

Comment:
I need to look into this a bit more. This doesn't feel right.

Reply:
Where is the check for `manufacturer and manufacturer code match`? Additionally, the name is not unique. Perhaps it should be searched by code and then compared to name?

Reply:
@awalkowiak manufacturer code given by user is not the same that "manufactrer.code" is. I'm still not sure how this should work. In exported data there is manufacturer's name given so I thought that users would always give name in this column.